runtime/lua/vim/lsp/handlers.lua

```diff
@@ -149,6 +149,32 @@ M['workspace/applyEdit'] = function(_, _, workspace_edit)
   }
 end
 
+--@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_configuration
+M['workspace/configuration'] = function(err, _, params, client_id)
+  local client = vim.lsp.get_client_by_id(client_id)
+  local client_name = client and client.name or string.format("id=%d", client_id)
+  if not client then
+    err_message("LSP[", client_name, "] client has shut down after sending the message")
+  end
```

Comment:
```suggestion
  if not client then
    err_message("LSP[id=", client_id, "] client has shut down after sending the message")
  end
```

`client_name` was only used in the `not client` case, so it always evaluates to `id=<client_id>`.

Reply:
We were discussing this a bit in https://github.com/neovim/neovim/pull/13652

I'm good with standardizing on reporting the error by client_id, especially because there can be multiple clients in the log sharing the same client_name

Reply:
My suggestion still keeps the same error reporting format as there was before - it just removed the `client and client.name` case which is dead code. `client.name` is never used. If the `client` is available then the `if` branch is skipped.

Reply:
Re-reading it now I see that we couldn't even get client_name if we wanted to for that error message. Thanks for the catch!